src/components/selectable-table/SelectableTable.js

```diff
@@ -0,0 +1,155 @@
+ * indicating those rows that should be initially selected.
+ * @prop {boolean} allowMultiple Whether to allow multiple rows to be selected. By default, false.
+ * @prop {boolean} allowUncheck Whether to allow selected rows to be un-checked. By default, false.
+ * @prop {boolean} showTableHead Whether to show the table header element. By default, true.
+ */
+export default function SelectableTable(props) {
+  const {
+    columns,
+    data,
+    onChange,
+    idKey = 'id',
+    valueKey = 'value',
+    allowMultiple = false,
+    allowUncheck = false,
+    showTableHead = true,
+  } = props;
+
+  const [selectedRows, setSelectedRows] = useState(null);
+
+  // Callback function to update the `selectedRows` state.
+  const onSelectRow = useCallback((value, checked) => {
+    if (!allowMultiple) {
+      if (checked) {
+        setSelectedRows([value]);
+      } else if (!checked && allowUncheck) {
+        setSelectedRows([]);
+      }
+    } else if (checked) {
+      setSelectedRows(union(selectedRows || [], [value]));
+    } else if (!checked && allowUncheck) {
+      setSelectedRows(difference(selectedRows || [], [value]));
+    }
+  }, [allowMultiple, allowUncheck, selectedRows]);
+
+  // Handler for checkbox input elements.
+  const handleInputChange = useCallback((event) => {
+    const { target } = event;
+    const { checked } = target;
+    const { value } = target;
+    onSelectRow(value, checked);
+  }, [onSelectRow]);
+
+  // Function to map row IDs to corresponding objects
+  // to pass to the `onChange` callback.
+  const getDataFromIds = useCallback(ids => ids.map(id => ({
+    [idKey]: id,
+    data: data.find(item => item[idKey] === id),
+  })), [data, idKey]);
+
+  // Function to check if a row ID has been selected.
+  const isSelected = useCallback(id => (
+    Array.isArray(selectedRows) && selectedRows.includes(id)
+  ), [selectedRows]);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    // Check whether an initial set of rows should be selected.
+    const initialSelectedRows = data
+      .map((d) => {
+        if (d[valueKey] !== undefined && d[valueKey]) {
+          return d[idKey];
+        }
+        return null;
+      })
+      .filter(Boolean);
+    if (initialSelectedRows.length > 0 && !isEqual(initialSelectedRows, selectedRows)) {
+      setSelectedRows(initialSelectedRows);
+    }
+  }, [data, idKey, valueKey]);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    // Call the `onChange` prop function with an updated row or set of rows.
+    if (!onChange || !selectedRows) {
+      return;
+    }
+    const selectedRowData = getDataFromIds(selectedRows);
+    if (!allowMultiple && selectedRows.length === 1) {
+      onChange(selectedRowData[0]);
+    } else if (!allowMultiple && selectedRows.length === 0) {
+      onChange(null);
+    } else if (allowMultiple) {
+      onChange(selectedRowData);
+    }
```

Comment:
```suggestion
    if (allowMultiple) {
      onChange(selectedRowData);
    } else if (selectedRows.length === 1) {
      onChange(selectedRowData[0]);
    } else if (selectedRows.length === 0) {
      onChange(null);
    }
```

I might be missing something.

Reply:
No this looks correct, thanks!